File: app/src/main/java/com/tonitealive/app/ui/views/BaseActivity.java
package com.tonitealive.app.ui.views;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.tonitealive.app.ToniteAliveApplication;
import com.tonitealive.app.internal.di.ComponentFactory;
import com.tonitealive.app.internal.di.components.ApplicationComponent;


public abstract class BaseActivity extends AppCompatActivity {

    protected ComponentFactory getComponentFactory() {
        ToniteAliveApplication application = (ToniteAliveApplication) getApplication();
        return application.getComponentFactory();
    }

    protected ApplicationComponent getApplicationComponent() {
        ToniteAliveApplication application = (ToniteAliveApplication) getApplication();
        return application.getApplicationComponent();
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initInjector();
    }

    protected abstract void initInjector();

}

File: app/src/main/kotlin/com/tonitealive/app/ui/BaseActivity.kt
package com.tonitealive.app.ui

import android.os.Bundle
import android.os.PersistableBundle
import android.support.v7.app.AppCompatActivity
import com.tonitealive.app.ToniteAliveApplication
import com.tonitealive.app.internal.di.ComponentHolder
import com.tonitealive.app.internal.di.modules.ActivityModule


abstract class BaseActivity : AppCompatActivity() {

    protected val applicationComponent by ComponentHolder {
        (application as ToniteAliveApplication).applicationComponent
    }

    protected val activityModule: ActivityModule
        get() = ActivityModule(this)

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        applicationComponent.inject(this)
    }


}